src/main/java/org/testng/IReporter2.java

```diff
@@ -0,0 +1,18 @@
+package org.testng;
+
+import org.testng.reporters.XMLReporterConfig;
+import org.testng.xml.XmlSuite;
+
+import java.util.List;
+
+/**
+ * This interface can be implemented by clients to generate a report.  Its method
+ * generateReport() will be invoked after all the suite have run and the parameters
+ * give all the test results that happened during that run.
+ */
+public interface IReporter2 extends ITestNGListener {
+  /**
+   * Generate a report for the given suites with specified attributes
+   */
+  void generateReport(List<XmlSuite> xmlSuites, List<ISuite> suites, XMLReporterConfig config);
```

Comment:
Rename XMLReporterConfig to ReporterConfig

Reply:
I think XMLReporterConfig is more suitable. It has more useful options that really needed in reporter classes

Reply:
All reporters are not xml, so it makes no sense to have a xml related param for a not xml reporter.